mipengine/controller/node_landscape_aggregator.py

```diff
@@ -0,0 +1,287 @@
+import asyncio
+import json
+import sys
+from typing import Dict
+from typing import List
+from typing import Tuple
+
+import dns.resolver
+from asgiref.sync import sync_to_async
+
+from mipengine.controller import DeploymentType
+from mipengine.controller import config as controller_config
+from mipengine.controller import controller_logger as ctrl_logger
+from mipengine.controller.celery_app import get_node_celery_app
+from mipengine.controller.common_data_elements import CommonDataElement
+from mipengine.controller.common_data_elements import CommonDataElements
+from mipengine.controller.data_model_registry import data_model_registry
+from mipengine.controller.node_registry import node_registry
+from mipengine.node_exceptions import IncompatibleCDEs
+from mipengine.node_info_DTOs import NodeInfo
+
+NODE_LANDSCAPE_AGGREGATOR_REQUEST_ID = "NODE_LANDSCAPE_AGGREGATOR"
+# TODO remove import get_node_celery_app, pass the celery app  (inverse dependency)
+# so the module can be easily unit tested
+
+logger = ctrl_logger.get_background_service_logger()
+
+GET_NODE_INFO_SIGNATURE = "mipengine.node.tasks.common.get_node_info"
+GET_NODE_DATASETS_PER_DATA_MODEL_SIGNATURE = (
+    "mipengine.node.tasks.common.get_node_datasets_per_data_model"
+)
+GET_DATA_MODEL_CDES_SIGNATURE = "mipengine.node.tasks.common.get_data_model_cdes"
+NODE_LANDSCAPE_AGGREGATOR_UPDATE_INTERVAL = (
+    controller_config.node_registry_update_interval
+)
+
+CELERY_TASKS_TIMEOUT = controller_config.rabbitmq.celery_tasks_timeout
+
+
+def _get_nodes_addresses_from_file() -> List[str]:
+    with open(controller_config.localnodes.config_file) as fp:
+        return json.load(fp)
+
+
+def _get_nodes_addresses_from_dns() -> List[str]:
+    localnodes_ips = dns.resolver.query(controller_config.localnodes.dns, "A")
+    localnodes_addresses = [
+        f"{ip}:{controller_config.localnodes.port}" for ip in localnodes_ips
+    ]
+    return localnodes_addresses
+
+
+def _get_nodes_addresses() -> List[str]:
+    if controller_config.deployment_type == DeploymentType.LOCAL:
+        return _get_nodes_addresses_from_file()
+    elif controller_config.deployment_type == DeploymentType.KUBERNETES:
+        return _get_nodes_addresses_from_dns()
+    else:
+        return []
```

Comment:
Please add this logic in a separate file to make this module a bit smaller.

This logic is used to get the node addresses of the current landscape. Could be improved in the future with dependency injection.